src/serialization_support.cpp

```diff
@@ -0,0 +1,990 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rosidl_dynamic_typesupport/api/serialization_support_interface.h>
+#include <rosidl_dynamic_typesupport/types.h>
+
+#include <rcutils/types/uint8_array.h>
+#include <rosidl_dynamic_typesupport_fastrtps/identifier.h>
+#include <rosidl_dynamic_typesupport_fastrtps/serialization_support.h>
+
+#include <fastrtps/types/DynamicDataFactory.h>
+#include <fastrtps/types/DynamicTypeBuilderFactory.h>
+
+#include "detail/fastrtps_dynamic_data.hpp"
+#include "detail/fastrtps_dynamic_type.hpp"
+#include "detail/fastrtps_serialization_support.hpp"
+
+
+// =================================================================================================
+// SERIALIZATION SUPPORT IMPL
+// =================================================================================================
+rosidl_dynamic_typesupport_serialization_support_impl_t *
+rosidl_dynamic_typesupport_fastrtps_create_serialization_support_impl()
+{
+  auto serialization_support_impl = new rosidl_dynamic_typesupport_serialization_support_impl_t();
+  serialization_support_impl->library_identifier =
+    fastrtps_serialization_support_library_identifier;
+
+  auto serialization_support_impl_handle = new fastrtps__serialization_support_impl_handle_t();
+
+  // The actual business
+  serialization_support_impl_handle->type_factory_ =
+    eprosima::fastrtps::types::DynamicTypeBuilderFactory::get_instance();
+  serialization_support_impl_handle->data_factory_ =
+    eprosima::fastrtps::types::DynamicDataFactory::get_instance();
+
+  serialization_support_impl->handle = serialization_support_impl_handle;
+
+  return serialization_support_impl;
+}
+
+
+// =================================================================================================
+// SERIALIZATION SUPPORT INTERFACE
+// =================================================================================================
+rosidl_dynamic_typesupport_serialization_support_interface_t *
+rosidl_dynamic_typesupport_fastrtps_create_serialization_support_interface()
+{
+  auto serialization_support_interface =
+    new rosidl_dynamic_typesupport_serialization_support_interface_t();
```

Comment:
We discussed this too, that it should be a static variable and returned as `const *`. But this would be in the allocation refactor follow up.